devicetypes/zooz/zooz-zen32-scene-controller.src/zooz-zen32-scene-controller.groovy

```diff
@@ -0,0 +1,459 @@
+		fingerprint mfr:"027A", prod:"7000", model: "A008", deviceJoinName:"Zooz Switch" // Zooz ZEN32 Scene Controller
+	}
+
+	preferences {
+		configParams.each { name, param ->
+			if (param.options) {
+				input name, "enum",
+					title: param.title,
+					description: "Default: ${param.options[param.defaultVal]}",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					options: param.options
+			} else if (param.range) {
+				input name, "number",
+					title: param.title,
+					description: "${param.desc} - Default: ${param.defaultVal}",
+					required: false,
+					displayDuringSetup: false,
+					defaultValue: param.defaultVal,
+					range: param.range
+			}
+		}
+	}
+}
+
+def installed() {
+	log.debug "installed()..."
+	initialize()
+	state.firstRun = true
+}
+
+def updated() {
+	log.debug "updated()..."
+	initialize()
+
+	if (!state.firstRun) {
+		executeConfigure()
+	} else {
+		state.firstRun = false
+	}
+}
+
+void initialize() {
+	if (!device.currentValue("checkInterval")) {
+		sendEvent([name: "checkInterval", value: ((60 * 60 * 3) + (5 * 60)), displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID]])
+	}
+
+	buttons.each { btn ->
+		if (!findChildByButton(btn)) {
+			addChildButton(btn)
+		}
+	}
+}
+
+void addChildButton(Map btn) {
+	log.debug "Creating Button ${btn.btnNum}"
+	try {
+		addChildDevice(
+			"Zooz",
+			"Zooz ZEN32 Scene Controller Button",
+			"${device.deviceNetworkId}:${btn.btnNum}",
+			device.getHub().getId(),
+			[
+				completedSetup: true,
+				label: "Zooz Button ${btn.btnNum}",
+				isComponent: false
+			]
+		)
+	} catch(Exception e) {
+		log.warn "${e}"
+	}
+}
+
+void executeConfigure() {
+	List<String> cmds = []
+
+	if (!device.currentValue("switch")) {
+		cmds << switchBinaryGetCmd()
+	}
+
+	if (!device.currentValue("firmwareVersion")) {
+		cmds << versionGetCmd()
+	}
+
+	configParams.each { name, param ->
+		Integer storedVal = getStoredVal(name)
+		Integer settingVal = getSettingVal(name)
+		if ((storedVal == null) || (storedVal != settingVal)) {
+			if (settingVal != null) {
+				log.debug "Changing ${param.title}(#${param.num}) from ${storedVal} to ${settingVal}"
+				cmds << configSetCmd(param, settingVal)
+			}
+			cmds << configGetCmd(param)
+		}
+	}
+
+	if (cmds) {
+		sendHubCommand(cmds, 100)
+	}
+}
+
+Integer getSettingVal(String name) {
+	Integer value = safeToInt(settings[name], null)
+	if ((value == null) && (getStoredVal(name) != null)) {
+		return configParams[name].defaultVal
+	} else {
+		return value
+	}
+}
+
+Integer getStoredVal(String name) {
+	return safeToInt(state[name], null)
+}
+
+def ping() {
+	log.debug "ping()..."
+	return [ switchBinaryGetCmd() ]
+}
+
+def on() {
+	log.debug "on()..."
+	return [ switchBinarySetCmd(0xFF) ]
+}
+
+def off() {
+	log.debug "off()..."
+	return [ switchBinarySetCmd(0x00) ]
+}
+
+def refresh() {
+	log.debug "refresh()..."
+
+	initialize()
```

Comment:
Could You explain why do You call `initialize ` from refresh?

Reply:
@KKlimczukS  When I wrote the original custom DTH I ran into an issue with the child devices sometimes not getting created during inclusion and adding that line to the refresh command seemed to fix it.

It's possible that was a coincidence or it could have been a platform issue at that time because I wrote the original DTH a long time ago so removing it might not cause any problems, but I don't think it's hurting anything so is it OK if I leave it there?

I'll submit a PR with the other changes tonight.

Reply:
Please, could You double-check if there are still issues with creating child-devices ?
`refresh()` is not the best place for calling `initialize()` (unnecessary resource consuming - every time user refreshes the device while child devices are already created)
Did You try (for testing purposes) exposing a Boolean parameter to trigger this configuration from the UI?

Reply:
I've removed it.